File: SpringBootMultipleMavenModules/persistence/src/main/java/rc/persistence/DbSeeder.java
package rc.persistence;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import rc.domain.Hotel;

import java.util.ArrayList;
import java.util.List;

@Component
public class DbSeeder implements CommandLineRunner {
    private HotelRepository hotelRepository;

    public DbSeeder(HotelRepository hotelRepository){
        this.hotelRepository = hotelRepository;
    }

    @Override
    public void run(String... strings) throws Exception {
        Hotel marriot = new Hotel("Marriot", 5, true);
        Hotel ibis = new Hotel("Ibis", 3, false);
        Hotel goldenTulip = new Hotel("Golden Tulip", 4, true);

        List<Hotel> hotels = new ArrayList<>();
        hotels.add(marriot);
        hotels.add(ibis);
        hotels.add(goldenTulip);

        this.hotelRepository.save(hotels);
    }
}

File: SpringBootKotlinDemo/src/main/kotlin/rc/demo/DbSeeder.kt
package rc.demo

import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

/**
 * This class has a dependency on HotelRepository. We can declare that
 * dependency in the primary constructor and it will be injected
 * at runtime
 */
@Component
class DbSeeder(val hotelRepository: HotelRepository) : CommandLineRunner {
    override fun run(vararg p0: String?) {
        this.hotelRepository.deleteAll()

        val ibis = Hotel("Ibis", 3, 30)
        val intercontinental = Hotel("Intercontinental", 5, 90)
        val goldenTulip = Hotel(name="Golden Tulip", classification = 4, nbRooms = 40)

        val hotels = mutableListOf<Hotel>()
        hotels.add(ibis)
        hotels.add(intercontinental)
        hotels.add(goldenTulip)

        this.hotelRepository.save(hotels)
        println(" -- Database has been initialized")
    }
}